back-end/routes/recommendations.js

```diff
@@ -18,34 +18,99 @@ const getAccessToken = async (userId) => {
     return token ? token.token : null;
 };
 
-const fetchStitchAverages = async (songs, access_token) => {
+const fetchAudioFeatures = async (songs, access_token) => {
     const songIds = songs.map(song => (song.uri).split(':').pop());
     const stringOfIds = songIds.join(',');
+
     try {
         const response = await fetch(`https://api.spotify.com/v1/audio-features?ids=${encodeURIComponent(stringOfIds)}`, {
             headers: {
                 'Authorization': `Bearer ${access_token}`
             }
         });
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
         const audioData = await response.json();
+        return audioData.audio_features;
     } catch (error) {
         console.error(error);
+        return [];
     }
 };
 
+function calculateGlobalFeatureAverages(features) {
+    let totalValence = 0;
+    let totalEnergy = 0;
+    let totalInstrumentalness = 0;
+    let totalCount = 0;
+
+    features.forEach(feature => {
+        totalValence += feature.valence;
+        totalEnergy += feature.energy;
+        totalInstrumentalness += feature.instrumentalness;
+        totalCount++;
```

Comment:
Do we need the `totalCount`? Can't we just do `features.length` or anything else?